src/apps/Permissions/Permissions.js

```diff
@@ -65,244 +35,177 @@ class Permissions extends React.Component {
-
-                  <Screen position={1} animate={animateScreens}>
-                    {['app', 'entity'].includes(location.screen) && (
-                      <React.Fragment>
-                        {location.screen === 'app' && (
-                          <AppPermissions
-                            app={location.app}
-                            loading={appsLoading}
-                            address={location.address}
-                            onManageRole={this.handleManageRole}
-                          />
-                        )}
-                        {location.screen === 'entity' && (
-                          <EntityPermissions
-                            title="Permissions granted to this entity"
-                            loading={appsLoading || permissionsLoading}
-                            address={location.address}
-                          />
-                        )}
-                      </React.Fragment>
-                    )}
-                  </Screen>
-                </Wrap>
-              </AppLayout>
-
-              <AssignPermissionPanel
-                apps={apps}
-                opened={showAssignPermissionPanel}
-                onClose={this.closeAssignPermissionPanel}
-                wrapper={wrapper}
-              />
-
-              <ManageRolePanel
-                app={location.app}
-                apps={apps}
-                opened={managedRole !== null}
-                onClose={this.closeManageRolePanel}
-                role={managedRole}
+  }, [apps, params, onParamsRequest])
+
+  const location = getLocation(params, apps)
+
+  const managedRole =
+    location.screen === 'app' &&
+    location.app &&
+    location.secondaryScreen === 'role'
+      ? resolveRole(location.app.proxyAddress, location.secondaryData)
+      : null
+
+  return (
+    <React.Fragment>
+      <div
+        css={`
+          // This element is only used to reset
+          // the view scroll using scrollIntoView()
+          position: absolute;
+          top: 0;
+          left: 0;
+          width: 1px;
+          height: 1px;
+        `}
+        ref={scrollTopElement}
+      />
+
+      <Header
+        primary={
+          location.screen === 'app' && location.app ? (
+            <Header.Title>
+              {`${location.app.name} permissions`}
+              <LocalIdentityBadge
+                entity={location.app.proxyAddress}
+                shorten
+                css={`
+                  margin-left: ${2 * GU}px;
+                `}
               />
-            </React.Fragment>
+            </Header.Title>
+          ) : (
+            'Permissions'
           )
-        }}
-      </PermissionsConsumer>
-    )
-  }
+        }
+        secondary={
+          <Button
+            mode="strong"
+            onClick={createPermission}
+            label="New permission"
+            icon={<IconPlus />}
+            display={layoutName === 'small' ? 'icon' : 'label'}
```

Comment:
I'm not very fond of this api, for a button I'd find it more intuitive to set the `children` instead of a `label`/`icon` pair, and maybe instead of `display` we could use `compact` (like in other places) to take care of the paddings?

Reply:
Yes it’s a tricky one. It is still possible to use `children`, but if you use `label` / `icon`, you get certain things:

- `Button` will add some spacing between the two.
- When left as the default, the `display` will change automatically in certain cases, e.g. to move from a text only to a label only. Actually the first and only case we have at the moment is `Header:secondary`, which is exactly where we are now, so we can remove this prop :laughing: 

We could achieve the same thing by detecting that an icon has been inserted as the first position of `children`, but I like that these props make it explicit that icon and label are handled as separate things, making the magic a bit less surprising.

Reply:
No, no, I get the magic that happens, my comment is that if I'm using a `Button` component, I'd assume an interface on setting children instead of all of this, how can we make it more obvious to the devs that they have all of this - besides the readme, not sure I have an answer though.

Reply:
Yes that’s what I was addressing in the last part of my comment. People probably expect `children` to be the normal way to use `Button`, because it was the case before, and because of `<button>label</button>` in HTML (although `<input type="button" value="label" />` also exists :laughing:). But I think it’s better to stay explicit here, even if it makes the component slightly less intuitive. I’m also thinking that it might be a good idea to deprecate `children` altogether, to avoid any confusion.

/cc @sohkai